BruceDevices/firmware
# Feature/pwnagotchi_pwngrid_spam

#### Proposed Changes ####

I suggest another Pwnagotchi PR, more customized, with Pwngrid spam feature.

*This work mainly comes from @7h30th3r0n3, @viniciusbo and @sduenasg.*

#### Types of Changes ####

Added features:
* new Pwnagotchi option in WiFi menu;
* find and act as a Pwnagotchi friend via Pwngrid;
* run Pwngrid spam:
  * spam many Pwngrid names and faces all around;
  * [option] flood identities: uniq identifiers flooded in Pwngrid logs;
  * [option] run DoScreen: send very long names and faces which override Pwnagotchi screen.
* Pwngrid spam is set from a customizable configuration file (see commit in *sd_files/pwnagotchi/pwngridspam.txt*).

#### Verification ####

Pwnagotchi should be able to work on any device.  

#### Testing ####

Running Pwnagotchi should:
1. Wakeup the pwnagotchi;
2. Start acting as a friend;
3. Press *select* should display Pwnagotchi menu with:
  * Find frens --> return acting as a friend by sending simple Pwngrid payloads;
  * Pwngrid spam --> run Pwngrid spam: 
    * press *esc* to return acting as a friend
    * press *next* to enable / disable identity flood
    * press *select* to enable / disable DoScreen.
  * Main Menu --> return to main menu.

Demo here : https://youtu.be/PREiRrVc7Tg

#### User-Facing Change ####

Users has to put the config file at the root of the SD card / LittleFS like this:  
`/pwnagotchi/pwngridspam.txt`

If not, it will not load Pwngrid spam feature.
